plugins/inputs/interrupts/interrupts.go

```diff
@@ -90,14 +98,63 @@ scan:
 		} else if len(fields) > cpucount {
 			irq.Type = strings.Join(fields[cpucount+1:], " ")
 		}
+		if spurious {
+			file := filepath.Join("/proc/irq", irq.ID, "spurious")
+			f, err := os.Open(file)
+			if err == nil {
+				irq.HasSpurious, irq.SpuriousCount, irq.SpuriousUnhandled = parseSpurious(f)
+				_ = f.Close()
+			}
+		}
 		irqs = append(irqs, *irq)
 	}
 	if scanner.Err() != nil {
-		return nil, fmt.Errorf("Error scanning file: %s", scanner.Err())
+		return nil, fmt.Errorf("error scanning file: %s", scanner.Err())
+	}
+
+	// determine the rightmost CPU column with non-zero data
+	validCpuIndex := 0
+	for _, irq := range irqs {
+		var i int
+		for i = len(irq.Cpus) - 1; i > validCpuIndex && irq.Cpus[i] == 0; i-- {
+		}
+		if i > validCpuIndex {
+			validCpuIndex = i
+		}
+	}
+	// remove data for any CPUs above the validCpuIndex
+	validCpuCount := validCpuIndex + 1
+	for i := 0; i < len(irqs); i++ {
+		if len(irqs[i].Cpus) > validCpuCount {
+			irqs[i].Cpus = append(irqs[i].Cpus[:validCpuCount])
+		}
```

Comment:
Can you explain this hunk?  The comments are nice but they don't tell me the why behind it.

Reply:
Sure, it covers a case that not everyone will see in their distro, so it can be confusing if you've never seen it.